File: Python/2577_숫자의개수.py
if __name__ == '__main__':
	A = int(input())
	B = int(input())
	C = int(input())
	
	mul = str(A * B * C) # 세 수의 곱을 문자열로 바꿈
	
	arr = [0]*10 # 0~9까지의 수가 몇개 있는지 세기 위한 배열
	for m in mul:
		arr[int(m)] += 1 # 문자열이기 때문에 인덱스로 입력할 때는 int로 형변환 필요
	
	for i in arr:
		print(i)

File: C++/2577_숫자의개수.cpp
#include <iostream>
using namespace std;

int main() {
	int A, B, C, arr[10] = {0,};
	cin >> A >> B >> C;
	
	long long int mul; // 곱은 int의 범위를 초과할 수 있
	mul = A * B * C;
	while(mul != 0){ // 10으로 나눈 몫과 나머지를 이용하여 숫자를 한 자리씩 분리
		int idx; // 맨 뒷자리 숫자
		idx = mul%10; // 맨 뒷자리 숫자를 제거하고 남은 숫자
		mul /= 10;
		
		arr[idx]++; // 맨 뒤에 있는 숫자부터 세
	}
	
	for(int i = 0; i < 10; i++){
		cout << arr[i] << '\n';
	}
	
	return 0;
}